src/rules/restrictPlusOperandsRule.ts

```diff
@@ -40,6 +41,8 @@ export class Rule extends Lint.Rules.TypedRule {
     }
 }
 
+type AddTypes = "string" | "number" | "invalid";
```

Comment:
I don't think we need a new type for this; I would prefer the union inlined in two places. This change makes the code more confusing